### PR TITLE
NIP-25: reactions to a website

### PR DESCRIPTION
**Pull request by nikolat**
Define `kind 17` event to react to a website.
It includes an `r` tag with the website's URL.

I want to react to a website with a decentralized system.

#1380

### CODE REVIEW

**Review by fiatjaf — APPROVED** [review]

**Review by staab — APPROVED** [review]

File: 25.md [hunk 1]
```diff
@@ -52,6 +52,22 @@ func make_like_event(pubkey: String, privkey: String, liked: NostrEvent) -> Nost
...
+---------------------
+
+If the target of the reaction is a website, the reaction SHOULD be a `kind 17` event and SHOULD include an `r` tag with the website's URL.
+
+```json
+{
+  "kind": 17,
+  "content": "⭐",
+  "tags": [
+    ["r", "https://example.com/"]
```
Comment (vitorpamplona) [inline 25.md:65]: Normalize the URL, like on https://datatracker.ietf.org/doc/html/rfc3986#section-6. 

You might also want to add guidance on fragments (can you like a section of a page?)

Also, keep in mind that there is no `startsWith` filter in Nostr, which means that if you want to get all reactions to a domain name (or path), you have to create individual tags for each.

File: 25.md [hunk 2]
```diff
@@ -52,6 +52,22 @@ func make_like_event(pubkey: String, privkey: String, liked: NostrEvent) -> Nost
 }
 ```
 
+Reactions to a website
+---------------------
+
+If the target of the reaction is a website, the reaction SHOULD be a `kind 17` event and SHOULD include an `r` tag with the website's URL.
```
Comment (vitorpamplona) [inline 25.md:58]: SHOULD -> MUST

**Comment by nikolat** [comment]
Thanks for your review.
I have updated the points you indicated.

**Comment by nikolat** [comment]
Is there anything else that needs to be corrected?

**Comment by staab** [comment]
LGTM, just waiting for two client implementations to merge.

File: 25.md (hunk 1 above)
Comment (fiatjaf) [reply 25.md:65]: Can we add this? And some examples maybe?

**Comment by penpenpng** [comment]
@staab 

> just waiting for two client implementations to merge.

There are two implementations:

- https://nikolat.github.io/makibishi-demo/
- https://penpenpng.github.io/makibishi-component/

And some websites using them:

- https://jiftechnify.github.io/motherfucking-nostr-client/
- https://nostr.compile-error.net/
- https://koteitan.github.io/nostr-post-checker/
- https://bewith.bz/

**Pull request merged**